src/Omnipay/Common/Message/AbstractResponse.php

```diff
@@ -182,8 +183,9 @@ public function getRedirectResponse()
             throw new RuntimeException('This response does not support redirection.');
         }
 
+        /** @var $this RedirectResponseInterface */
         if ('GET' === $this->getRedirectMethod()) {
-            return HttpRedirectResponse::create($this->getRedirectUrl());
+            return new RedirectResponse($this->getRedirectUrl());
```

Comment:
Hari - there is no "proper PSR-7 response object". There is a PSR-7 interface, which a number of implementations extend. With all those implementations, you can use the PSR-7 interface methods, and use whatever additional methods those implementations provide.